src/handlers/platforms/aws-lambda/platform-adapter.ts

```diff
@@ -0,0 +1,110 @@
+import type {
+  APIGatewayProxyEventV2,
+  APIGatewayProxyStructuredResultV2,
+  Context,
+} from "aws-lambda";
+
+import {
+  ActionHandlerResult,
+  HTTPMethod,
+  PlatformAdapterInterface,
+} from "@/handlers/shared/generic-adapter-use-case-types";
+
+export type AwsLambdaHandlerInput = APIGatewayProxyEventV2;
+export type AWSLambdaHandler = (
+  event: APIGatewayProxyEventV2,
+  context: Context
+) => Promise<APIGatewayProxyStructuredResultV2>;
+
+/** PlatformAdapter for AWS Lambda HTTP events
+ *
+ * Platform adapters are used in Actions to handle generic request logic
+ * like getting body, headers, etc.
+ *
+ * Thanks to this Actions logic can be re-used for each platform
+
+ * @see {PlatformAdapterInterface}
+ * */
+export class AwsLambdaAdapter implements PlatformAdapterInterface<AwsLambdaHandlerInput> {
+  public request: AwsLambdaHandlerInput;
+
+  constructor(private event: APIGatewayProxyEventV2, private context: Context) {
+    this.request = event;
+  }
+
+  getHeader(requestedName: string): string | null {
+    // TODO: Check if it works correctly with both API gateway and Lambda Function URL
+
+    // Lambda headers are always in lowercase for new deployments using API Gateway,
+    // they use HTTP/2 which requires them to be lowercase
+    //
+    // if there are multiple values they are separated by comma: ,
+    // https://docs.aws.amazon.com/apigateway/latest/developerguide/set-up-lambda-proxy-integrations.html#api-gateway-simple-proxy-for-lambda-input-format
+    const name = requestedName.toLocaleLowerCase();
+    return this.request.headers[name] || null;
+  }
+
+  async getBody(): Promise<unknown | null> {
+    if (!this.request.body) {
+      return null;
+    }
+
+    return JSON.parse(this.request.body);
```

Comment:
question: Is it save to run `JSON.parse` without proper error handling in case `this.request.body` will has invalid json?

Reply:
Yes, we want it to throw error so that we know that request was invalid because of malformated JSON, not that the request was empty (if we would return null).

Ideally I would use [`neverthrow`](https://github.com/supermacro/neverthrow) so that we don't throw exceptions, but I didn't want to introduce another dependency ;)

Reply:
In places we use `getBody` we have `try...catch`